exporters/otlp/otlptrace/otlptracehttp/options.go

```diff
@@ -0,0 +1,164 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package otlptracehttp
+
+import (
+	"crypto/tls"
+	"time"
+
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace/internal/otlpconfig"
+)
+
+const (
+	// defaultMaxAttempts describes how many times the driver
+	// should retry the sending of the payload in case of a
+	// retryable error.
+	defaultMaxAttempts int = 5
+	// defaultTracesPath is a default URL path for endpoint that
+	// receives spans.
+	defaultTracesPath string = "/v1/traces"
+	// defaultBackoff is a default base backoff time used in the
+	// exponential backoff strategy.
+	defaultBackoff time.Duration = 300 * time.Millisecond
+)
+
+// Compression describes the compression used for payloads sent to the
+// collector.
+type Compression otlpconfig.Compression
+
+const (
+	// NoCompression tells the driver to send payloads without
+	// compression.
+	NoCompression = Compression(otlpconfig.NoCompression)
+	// GzipCompression tells the driver to send payloads after
+	// compressing them with gzip.
+	GzipCompression = Compression(otlpconfig.GzipCompression)
+)
+
+// Option applies an option to the HTTP driver.
+type Option interface {
+	applyHTTPOption(*otlpconfig.Config)
+}
+
+// RetrySettings defines configuration for retrying batches in case of export failure
+// using an exponential backoff.
+type RetrySettings otlpconfig.RetrySettings
+
+type wrappedOption struct {
+	otlpconfig.HTTPOption
+}
+
+func (w wrappedOption) applyHTTPOption(cfg *otlpconfig.Config) {
+	w.ApplyHTTPOption(cfg)
+}
+
+// WithEndpoint allows one to set the address of the collector
+// endpoint that the driver will use to send spans. If
+// unset, it will instead try to use
+// DefaultCollectorHost:DefaultCollectorPort. Note that the endpoint
+// must not contain any URL path.
+func WithEndpoint(endpoint string) Option {
+	return wrappedOption{otlpconfig.WithEndpoint(endpoint)}
+}
+
+// WithTracesEndpoint allows one to set the address of the collector
+// endpoint that the driver will use to send spans. If
+// unset, it will instead try to use the Endpoint configuration.
+// Note that the endpoint must not contain any URL path.
+func WithTracesEndpoint(endpoint string) Option {
```

Comment:
I'm starting to think that we do not need both `With*` and `WithTraces*` option.

It used to make sense because the same package would both, send metrics and traces. As these clients are separated by signal, all configs are about this specific signal. Environment variable will keep working, just we don't need to have two different options that does the *same* configuration.

I'll be updating this PR, and create one for cleaning up the `otlptracegrpc` options.